features/collection/loaders/index.ts

```diff
@@ -148,3 +156,87 @@ const collectionQuery = /* GraphQL */ `
       }
    }
 `;
+
+interface Post {
+   ID: number;
+   post_author: string;
+   post_date: string;
+   post_date_gmt: string;
+   post_content: string;
+   post_title: string;
+   post_excerpt: string;
+   permalink?: string;
+   featured_image: PostImage;
+   categories: PostCategory[];
+}
+
+interface PostCategory {
+   term_id: number;
+   name: string;
+   slug: string;
+   term_group: number;
+   term_taxonomy_id: number;
+   taxonomy: string;
+   description: string;
+   parent: number;
+   count: number;
+   filter: string;
+   cat_ID: number;
+   category_count: number;
+   category_description: string;
+   cat_name: string;
+   category_nicename: string;
+   category_parent: number;
+}
+
+interface PostImage {
+   thumbnail: string;
+   medium: string;
+   medium_large: string;
+   large: string;
+   '1536x1536': string;
+   '2048x2048': string;
+   'post-header': string;
+   sm: string;
+   md: string;
+   lg: string;
+   'homepage-featured': string;
+   'homepage-small': string;
+   'homepage-medium': string;
+   'rp4wp-thumbnail-post': string;
+}
+
+async function loadRelatedNews(tags: string[]): Promise<News[]> {
+   const base64Credentials = convertToBase64(
+      `${WP_BASIC_AUTH_USER}:${WP_BASIC_AUTH_PASSWORD}`
+   );
+   const response = await fetch(`${WP_ORIGIN}/wp-json/wp/v2/posts/related`, {
+      method: 'POST',
+      body: JSON.stringify({ tags }),
+      headers: {
+         Authorization: `Basic ${base64Credentials}`,
+         'Content-Type': 'application/json',
+      },
+   });
+   if (response.status >= 200 && response.status < 300) {
+      const rawStories: Post[] = await response.json();
+      return rawStories.map<News>((rawStory) => {
+         return {
+            id: rawStory.ID,
+            title: rawStory.post_title,
+            date: rawStory.post_date,
+            category: rawStory.categories[0]?.name,
+            image: rawStory.featured_image
+               ? { url: rawStory.featured_image.md }
+               : null,
+            permalink: rawStory.permalink || '',
```

Comment:
Do these end up being correct? When our app embeds the WP content into our site (we access the API from the backend) we post-process the response and rewrite the urls (among other things). Do these urls make sense given the meant-to-be-private wordpress site?

I think we'd probably want this code to be pinging the equivalent API on our site so we can do the post-processing on the response. Thoughts @jarstelfox ?

Reply:
> I think we'd probably want this code to be pinging the equivalent API on our site so we can do the post-processing on the response. Thoughts @jarstelfox ?

That was my immediate thought.

I think we will need to set up an API on our side to proxy the data. CC @Annika29 @MilesRobinson

Reply:
@jarstelfox that would be great, so we bypass basic auth and CORS issues 👍

Reply:
Awesome, I’ll see if I can get to it tomorrow or Monday so I can unblock you. Sorry about that!

Reply:
Sorry about the delay. I have a pull open now.

Reply:
Hi all I forgot to say this is working and live: https://ifixit.com/api/2.0/related_posts?data={%20%22tags%22:%20[%20%22repair%22%20],%20%22per_page%22:%205%20}

just hit https://ifixit.com/api/2.0/related_posts with `?data=<some_valid_json>`